.github/workflows/ghpages.yml

```diff
@@ -0,0 +1,46 @@
+name: CI
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v1
+    
+    - name: Print GitHub context
+      env:
+        GITHUB_EVENT: ${{ github.event_name }}
+        GITHUB_REF: ${{ github.ref }}
+      run: |
+        echo $GITHUB_EVENT $GITHUB_REF
+    
+    - name: Generate Doxygen Website
+      if: github.event_name == 'push' && (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/devel')
+      run: |
+        cd ${GITHUB_WORKSPACE}
+        mkdir deploy
+        sudo apt-get install build-essential doxygen graphviz
+        cd ${GITHUB_WORKSPACE}
+        git clone https://github.com/robotology/idyntree
+        cd idyntree
+        mkdir build
+        cd build 
+        cmake -DIDYNTREE_ONLY_DOCS:BOOL=ON ..
+        make dox
+        cp -r ./doc/html ${GITHUB_WORKSPACE}/deploy/master
+        git checkout devel
+        cmake -DIDYNTREE_ONLY_DOCS:BOOL=ON ..
+        make dox
+        cp -r ./doc/html ${GITHUB_WORKSPACE}/deploy/devel
+
+    - name: Deploy
+      if: github.event_name == 'push' && (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/devel')
+      uses: JamesIves/github-pages-deploy-action@master
+      env:
+        ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
```

Comment:
Just to be sure, is the default token not enough to trigger ghpages rebuild? For another project I have a staging branch for a very similar deployment and I still have this doubt.

Reply:
I tried to use `GITHUB_TOKEN`, but in that case the build if failing:
![problems](https://user-images.githubusercontent.com/1857049/69098085-2907d380-0a58-11ea-8aa5-edfd9eed8c0d.png)
I reverted to use an explicit `ACCESS_TOKEN` , and everything  is working fine. So I guess `GITHUB_TOKEN` is still not sufficient.

Reply:
Ok, thanks for the test. If I recall well, @drdanz had a similar problem with the token. Not sure though if it was the same.

Reply:
There seems to be a lot of dicussions on this: 
* https://github.com/JamesIves/github-pages-deploy-action/issues/5
* https://github.community/t5/GitHub-Actions/Github-action-not-triggering-gh-pages-upon-push/m-p/31266/highlight/true#M743

Reply:
My problem was a little different... The `GITHUB_TOKEN` does not allow to push 1) on PR branches from forks and 2) on protected branches.
Also I remember something about pushes using `GITHUB_TOKEN` that do not trigger the deployment of gh-pages, but I'm not very sure about this